### ISSUE TITLE
Add a feedback form

### ISSUE THREAD

**Issue by Riddhi162** [Innovateninjas/Paws-frontend#278]
Adding a feedback form provides an overall customer satisfaction and the organisation also knows about it.
It may include
1.Overall experience out of five stars
2.Customers name phone no and email address 
3.Any improvements by the customer

**Comment by codewarnab** [comment]
> Adding a feedback form provides an overall customer satisfaction and the organisation also knows about it. It may include 1.Overall experience out of five stars 2.Customers name phone no and email address 3.Any improvements/suggestions recommended by the customer

seems a good feature share your ideas/thoughts here if you are willing to work on it where will you like it to be added and how the ui will look like any rough design or some similar scrrenshot lets discuss below

**Comment by Riddhi162** [comment]
There are multiple options 
1.we can add it in the bottom bar where there are options like register,home etc 
2.After the user reports the pet we can do something like after an hour or so he/she gets a notification and it redirects to the feedback page 
3.we can create it as a pop-up and position it somewhere on the website
4.Or else we create a footer only where we have the whole feedback section the title and icon of the organisation

**Comment by Riddhi162** [comment]
So is this issue assigned to me can I work on it?

**Comment by Riddhi162** [comment]
> After successfully reporting an animal, users should receive a notification asking for feedback. This feedback request should be sent only once, specifically during the first time a report is made. The feedback form should include a star rating system and a text box for additional comments. The collected feedback data should be stored in the database. Additionally, an interface should be created for viewing all feedback entries.
> 
> **will be treated as three separate issues** _(EACH OF LEVEL 3 GSSOC'24)_ Comment below in which part are you willing to work on
> 
> 1. **Feedback Page - User Side:**
>    
>    * Develop a new page or modal for users to provide feedback after reporting an animal.
>    * Design the feedback form with a star rating system and a text box for comments.
>    * Ensure the feedback request is triggered only once for the initial report.
> 2. **Admin Interface for Viewing Feedbacks:**
>    
>    * Create an admin interface to view all feedback entries.
>    * Implement a system to restrict access to the feedback interface to admins only.
>    * Design the interface to display feedback entries with relevant information such as user ratings and comments.
> 3. **Sending Notification to Users for Feedback:**
>    
>    * Integrate a notification system to send feedback requests to users after reporting an animal.
>    * Ensure that the notification is triggered only once for the initial report.
>    * Customize the notification content to encourage users to provide feedback.
> 
> ### Links:
> * Backend Repository: [Link to Backend Repository](https://github.com/Innovateninjas/Paws-Backend)

i would like to work on issue 1 and issue 3

**Comment by codewarnab** [comment]
@Riddhi162 ok  before start coding it would be better if you show us some screenshots or maybe some rough designs though figma/excalidraw  and note that pr for the ui may stay on hold until the neccsary changes i made by some other contributor fot accomodating this pr 
if you are ok with that i can assign you

**Comment by Riddhi162** [comment]
I'm ok with that i will provide some designs in a day or two n we can go ahead

**Comment by codewarnab** [comment]
> I'm ok with that i will provide some designs in a day or two n we can go ahead

done take your time no hurry

**Comment by codewarnab** [comment]
> I'm ok with that i will provide some designs in a day or two n we can go ahead

its been a week! Give update

**Comment by Riddhi162** [comment]
yes sure i will provide the form design till evening sorry for being late.

**Comment by shilpashingnapure** [comment]
@codewarnab is issue 2 assigned to anyone , if not i would like to work on

**Comment by codewarnab** [comment]
> @codewarnab is issue 2 assigned to anyone , if not i would like to work on

okay it would be great .
you can also discuss with @Riddhi162  as it part of a same feature

**Comment by Riddhi162** [comment]
Please checkout this form if it looks good.I have now just made the design work related to backend will be done once you approve this.Thanks
![image](https://github.com/Innovateninjas/Paws-frontend/assets/135015856/53776fc9-c11c-4ea9-a555-199c0d0fbaad)

**Comment by shilpashingnapure** [comment]
Hi , 
I'm little confused on this issue for how feedbacks will visible to whom , please let me correct if i'm understand this wrong 
1. who is admin here (ngo is admin or what) ?
2. if yes , then  do we have to handling the user feedback related to that ngo only or what ?

And in different perspective I have understand 
1. is this for app feedback ?
2. so user and ngo will give feedback. and admin will see all the feedback 
3. if yes , then  admin is django interface right  or we have admin ?

**Comment by codewarnab** [comment]
@shilpashingnapure   here admins mean owner of the app its actually feedback for whole app we will send notification to the user after reporting an animal alternately we can take review from ngos also but lets first build this initial system will see other things in future

**Comment by shilpashingnapure** [comment]
@codewarnab okay , got it
Thank you!!

**Comment by rishicds** [comment]
> @rishicds  looks good ?
> 

Yess it's great

**Comment by shilpashingnapure** [comment]
Hi @Riddhi162 , can you discussed me with how your planning to handle backend and all , so I also get some idea on this.

**Comment by Riddhi162** [comment]
> Hi @Riddhi162 , can you discussed me with how your planning to handle backend and all , so I also get some idea on this.

So as Arnab said we will be triggering this page only after the animal is reported. So in the feedback page there will be two responses and those will be stored 
1.maybe with the registered or login name of the user so we know who responded
2.Or maybe we can only store the rating and suggestion if we dont want to store the name of it along.
What are you thinking of this ?

**Comment by shilpashingnapure** [comment]
i think , it will good storing of user name also.

**Comment by Riddhi162** [comment]
so will we fetch it like if someone reports an animal her/his username might be stored right so should we store fetch that only or should i create a diff field for it in feedback form

**Comment by shilpashingnapure** [comment]
I mean while storing into database , in feedback form we no need to add

**Comment by codewarnab** [comment]
whats the update guys ?

**Comment by shilpashingnapure** [comment]
hey @codewarnab , actually I'm busy with some other stuffs , once that get done , i'll start to work on

**Comment by Riddhi162** [comment]
I have designed the feedback form. I have a question that suppose a same user reports animals again and again then will she/he be shown the feedback form every time there is a new report.
